### ISSUE TITLE
Flatting and labeling Images using R

### ISSUE THREAD

**Issue by KafeelBasha** [rstudio/keras#379]
Hello
I have a train folder in which I have sub-folders "Cats" and "Dogs", with number of images in each folder.

I would like to import images in sub-folders, flatten them and label accordingly using R. Trying to build a classifier using CNN.

Any example or reference will be helpful.

**Comment by jjallaire** [comment]
https://tensorflow.rstudio.com/blog/keras-image-classification-on-small-datasets.html

**Comment by KafeelBasha** [comment]
Thanks.
How about if I have 10 different sub-folders in Train folder.

**Comment by KafeelBasha** [comment]
I have a path "C:/ML/Classification/Image Data/" which has sub folder "class1", "class2", "class3" and "class4". And I am trying to import images from each sub folder using readImage() function from "EBImage" package in R(External source), also trying to label flatten image pixel rows by its class name, like images belongs to sub folder "class1" as "class1".

I was trying to use list.names() with pattern=".png" and list.dirs(), need some help.

**Comment by KafeelBasha** [comment]
I tried with three sub-folder, with below logic I am able to get desired result.

path="C:\\Users\\Kafeel Basha\\Desktop\\R Files\\Materials\\Import\\"

images=list.files(path,recursive=TRUE,pattern=".png",full.names=T)
pixel<-""
readimg<-function(files)
{
  img<-readImage(files)@.Data
 }

data<-sapply(images,readimg)
class(data)
pv=data.frame(t(data))
dim(pv)
library(dplyr)
rownames(pv)
pv$class=sapply(strsplit(rownames(pv),"/"),"[",2)
pv$class
